src/main/java/it/gov/pagopa/payhub/auth/service/a2a/AuthorizeClientCredentialsRequestService.java

```diff
@@ -3,24 +3,54 @@
 import it.gov.pagopa.payhub.auth.exception.custom.ClientUnauthorizedException;
 import it.gov.pagopa.payhub.auth.mapper.ClientMapper;
 import it.gov.pagopa.payhub.model.generated.ClientDTO;
+import org.springframework.beans.factory.annotation.Value;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 @Service
 @Slf4j
 public class AuthorizeClientCredentialsRequestService {
+	private static final String ERROR = "Unauthorized client for client-credentials";
+	private static final String REGEX = "^(\\w+\\s*)(piattaforma-unitaria\\b)$";
 	private final ClientService clientService;
 	private final ClientMapper clientMapper;
+	private final String clientSecretEnv;
 
-	public AuthorizeClientCredentialsRequestService(ClientService clientService, ClientMapper clientMapper) {
+	public AuthorizeClientCredentialsRequestService(
+			@Value("${piattaforma-unitaria-client-secret}") String clientSecretEnv,
+			ClientService clientService,
+			ClientMapper clientMapper) {
 		this.clientService = clientService;
 		this.clientMapper = clientMapper;
+		this.clientSecretEnv = clientSecretEnv;
 	}
 
 	public ClientDTO authorizeCredentials(String clientId, String clientSecret) {
+		Matcher matcher = Pattern.compile(REGEX).matcher(clientId);
+		if (matcher.matches()) {
```

Comment:
I found hard to read and interpret this code, let's give name based on the purpose, not base on what's done (this is a business class, let's use business names)

```
if(isPiattaformaUnitariaClientId(clientId){
   authorizePiattaformaUnitariaCredentials(...)
} else {
   authorizeSilCredentials(...)
}
```